test/preview/components/test_prompt.py

```diff
@@ -0,0 +1,59 @@
+import pytest
+
+import haystack
+from haystack.preview.components.prompt.prompt import Prompt
+
+from haystack.preview.components.prompt.models.base import prompt_model
+
+
+@prompt_model
+class MockImplementation:
+    def __init__(self, model_name_or_path: str, **kwargs):
+        pass
+
+    def invoke(self, *args, **kwargs):
+        return ["test response"]
+
+    @classmethod
+    def supports(cls, model_name_or_path: str, **kwargs) -> bool:
+        return model_name_or_path == "mock_model"
+
+
+@pytest.fixture()
+def mock_implementation(monkeypatch):
+    monkeypatch.setattr(haystack.preview.components.prompt.prompt, "IMPLEMENTATION_MODULES", [__name__])
+
+
+@pytest.mark.unit
+def test_prompt_custom_template_no_inputs(mock_implementation):
+    pn = Prompt(custom_template="This is my own template", model_name_or_path="mock_model")
```

Comment:
In which situation would we use a template without inputs?

Reply:
I cant' foresee that, but it's such a cheap test for a corner case that I don't see why not testing it :grin: I had in mind some pre-defined classification tasks, like sentiment classification, but it's really just for completeness.